tensorboard/components/plugin_util/plugin-host.ts

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {IPC, Message, MessageType} from './message.js';
+
+class HostIPC extends IPC {
+  sendMessage(iframe: HTMLIFrameElement, type: MessageType, payload: any): Promise<any> {
+    return this._sendMessage(iframe.contentWindow, type, payload);
+  }
+}
+
+const hostIPC = new HostIPC();
+const _listen = hostIPC.listen.bind(hostIPC);
+const _unlisten = hostIPC.unlisten.bind(hostIPC);
+
+/**
+ * payload must be JSON serializable.
+ */
+function _sendMessage(
+  iframes: HTMLIFrameElement[],
+  type: MessageType,
+  payload: any,
+): Promise<void> {
+  const sendMessageP = iframes.map((iframe) => {
```

Comment:
nit: I kind of prefer letting clients do the mapping, and having this take a single iframe